recipes/cpr/all/conanfile.py

```diff
@@ -60,8 +65,9 @@ def configure(self):
         if not self._supports_winssl:
             del self.options.with_winssl
 
-        if self._supports_openssl and tools.is_apple_os(self.settings.os):
-            self.options.with_openssl = False # Default libcurl in CCI is `with_ssl="darwin"` which is unclear if cpr supports this
+        if self.options.get_safe("with_openssl", False) and tools.is_apple_os(self.settings.os):
+            # https://github.com/whoshuu/cpr/issues/546
+            raise ConanInvalidConfiguration("cpr cannot be built on macOS with openssl")
```

Comment:
Not a fan of this diff since it removes support for older version which do work.

```py
  if tools.Version(self.version) < "1.6.0"
      # Do the old way
  else
      # Is does not work
```

Reply:
Re: moving into a validate method, sure. 

Re: the `InvalidConfiguration` this option would just disable openssl if you were on a Mac. Though I could add the old one back, would it make a difference? (I'm not 100% if `with_openssl` would have shown as a valid package option before, but it being silently disabled is bad if it did).

Reply:
@prince-chrismc Ok, things have been moved into a validate method, and those should be building now. I did just want to confirm with you about the disabling of SSL on apple OS's, though, given what I mentioned above. Would we still want to gate it off like you describe?

Reply:
Let's get a build to pass, I ll check into the new version and see why MacOS fails and come back with a good answer 😄 

I didnt see anything that would break it so I am a little suspicious I but need to double check and test it